Bottom: 791459c1708fc05fce36bf9d7c24a818e6760291
Top:    5e1a5948e8e23c4df713205764e300f2a9103dd5
Author: Christopher Voltz <cjunk@voltz.ws>
Date:   2016-04-29 02:49:49 -0500

Add tag, tag_col, tag_row keywords

Add new keywords:
  - tag(name) Associate the current row and column with name
  - tag_col(name) Return the column for the specified tag
  - tag_row(name) Return the row for the specified tag


---

diff --git a/script.js b/script.js
index 150fe3f..5c82ebc 100644
--- a/script.js
+++ b/script.js
@@ -113,6 +113,30 @@ function row() {
     return tablecalc_crow;
 }
 
+function tag(name) {
+    if (typeof tablecalc_tags[name] == "undefined") {
+        tablecalc_tags[name] = { row: row(), column: col() }
+    }
+
+    return "";
+}
+
+function tag_col(name) {
+    if (typeof tablecalc_tags[name] == "undefined") {
+        return 0;
+    }
+
+    return tablecalc_tags[name].column;
+}
+
+function tag_row(name) {
+    if (typeof tablecalc_tags[name] == "undefined") {
+        return 0;
+    }
+
+    return tablecalc_tags[name].row
+}
+
 function cell(x, y) {
     var tmp = tablecalcVal(x, y, tablecalc_table);
     if (tmp == "notset" || tmp == "notnum") {
@@ -287,6 +311,10 @@ function tablecalc(divID, formula, final) {
     if (typeof tablecalc_table === "undefined") {
         window.tablecalc_table = null;
     }
+    if (typeof tablecalc_tags === "undefined") {
+        window.tablecalc_tags = null;
+    }
+
     var oFormula = formula;
     if (isNaN(final)) {
         final = 0;
@@ -474,4 +502,4 @@ function findParentNodeByName(pNode, st) {
 
 function stripHTML(oldString) {
     return oldString.replace(/<[^>]*>/g, "");
-}
\ No newline at end of file
+}
diff --git a/syntax.php b/syntax.php
index 10dca49..5e4dcff 100644
--- a/syntax.php
+++ b/syntax.php
@@ -49,6 +49,9 @@ class syntax_plugin_tablecalc extends DokuWiki_Syntax_Plugin {
         $signs = "-~=+*.,;\/!|&\(\)";
         $pattern = "/[$signs]*([a-zA-Z]+)\(/is";
         $aAllowed = array(
+            "tag",
+            "tag_row",
+            "tag_col",
             "cell",
             "row",
             "col",
